### ISSUE TITLE
Getting Null on Patch Requests

### ISSUE THREAD

**Issue by amarwadi** [KevinDockx/JsonPatch#66]
Hello,
Is there anything special I need to do other than adding the Nuget Package and using the JsonPatchDocument<T> on the [FromBody] attribute for the HttpPatch methods on Web API? For some reason, no matter what I do, I always get a null object. Here's how my call looks like:

         [Route("sites/{siteId}/affiliates/{affiliateId}/codes/{codeId}/commissions/{id}"), HttpPatch,
         ResponseType(typeof(BoolResult)), ValidateSiteId()]
        public async Task<IHttpActionResult> AddTieredCommission(long siteId, long affiliateId, string codeId,
           string id, [FromBody] JsonPatchDocument<AddTieredCommissionModel> model)
        {
            
            return Ok();
        }

I am using ASP.NET Web API (not on ASP.NET Core). Any help would be appreciated.

Anup

**Comment by LouisChiero** [comment]
Are you setting the request's Content-Type as "application/json-patch+json"?

**Comment by Swoogan** [comment]
I had this same issue. Make sure that the JSON body of your request is formatted correctly. I was using RestSharp, which serializes using a formatter that doesn't respect the [SerializeAs] attributes on JsonPatchDocument. Using fiddler I was able to see what was being passed in the request body and with Insomnia I could confirm that it didn't work. By serializing with Newtsoft.Json, I got out put that looked like the json patch RFC.

**Comment by KevinDockx** [comment]
Closing issue as it seems to be solved.  If you experience additional problems, feel free to reopen.

**Comment by dpescatore** [comment]
How this was resolved? I have the same problem. 
I've tried adding 
`config.Formatters.JsonFormatter.SupportedMediaTypes.Add(
                new MediaTypeHeaderValue("application/json-patch+json"));`
And both with simple "application/json" but I always have a null object on server.

Here's signature of my api method

`public IHttpActionResult PatchEntity(int id, [FromBody]JsonPatchDocument<Entity> entity)`

and here my simple JSONPatch sent by restlet client. 

`{ "op": "replace", "path": "/seconds", "value": "12" }`

Any help on this?

Thanks,
Daniele
